R/model_rent_assistance.R

```diff
@@ -0,0 +1,78 @@
+
+#' @export 
+#' 
+#' @examples
+#' sample<- CJ(rent = 1:500, n_dependants = 0:3, has_partner = 0:1 > 0, is_homeowner = 0:1 > 0, lives_in_sharehouse = 0:1 > 0) 
+#' model_rent_assistance(sample, .Prop_rent_paid_by_RA = 0.75, Max_rate = 500, Min_rent = 100)
+
+model_rent_assistance <- function(sample_file,
+                                  baseline_fy = NULL,
+                                  baseline_Date = NULL,
+                                  .Prop_rent_paid_by_RA,
+                                  Max_rate = NULL,
+                                  Min_rent = NULL,
+                                  calc_baseline_ra = TRUE,
+                                  return. = "new_ra") {
+  
+  sample_file <- copy(sample_file)
+  
+  #check sample file has correct format
+  stopifnot(is.data.frame(sample_file))
+  if(!is.data.table(sample_file)) {
+    sample_file <- data.table(sample_file)
+  }
+  cols_required <- c("rent",  "n_dependants", "has_partner", "is_homeowner", "lives_in_sharehouse")
+  if (!all(cols_required %chin% colnames(sample_file))) {
+    absent_cols <- setdiff(cols_required, colnames(sample_file))
+    stop("`sample_file` lacked the following required columns:\n\t",
+         paste0(absent_cols, collapse = "\n\t"), ".\n")
+  }
+  
+  #actual calculation
+  Rent <- sample_file[['rent']]
+  N_dependants <- sample_file[['n_dependants']]
+  Has_partner <- sample_file[['has_partner']]
+  Is_homeowner <- sample_file[['is_homeowner']]
+  Lives_in_sharehouse <- sample_file[['lives_in_sharehouse']]
+  
+  if (calc_baseline_ra){
+    baseline_ra <- rent_assistance(fortnightly_rent = Rent, 
+                                   fy.year = baseline_fy,
+                                   Date = baseline_Date,
+                                   n_dependants = N_dependants,
+                                   has_partner = Has_partner,
+                                   is_homeowner = Is_homeowner,
+                                   lives_in_sharehouse = Lives_in_sharehouse)
+    
+    switch(return.,
+           "sample_file" = set(sample_file, j = "baseline_ra", value = baseline_ra),
+           "sample_file.int" = set(sample_file, j = "baseline_ra", value = as.integer(baseline_ra)))
+    
+  }
+  
+  ra <- rent_assistance(fortnightly_rent = Rent, 
+                        n_dependants = N_dependants,
+                        has_partner = Has_partner,
+                        is_homeowner = Is_homeowner,
+                        lives_in_sharehouse = Lives_in_sharehouse,
+                        .prop_rent_paid_by_RA = .Prop_rent_paid_by_RA,
+                        max_rate = Max_rate,
+                        min_rent = Min_rent)
+  
+  switch(return.,
+         "ra" = new_ra,
+         "sample_file" = set(sample_file, j = "new_ra", value = ra),
+         "sample_file.int" = set(sample_file, j = "new_ra", value = as.integer(ra)))
+  
+  eval(parse(text = return.))
```

Comment:
Couldn't figure out why it wasn't returning the `return.` value at the `switch()` level, so added the `eval(parse())` which probably isn't the best but it works for now.

Reply:
Ha you're right to be suspicious of `eval(parse` calls. The result is returning, but only invisibly. You can add `[]` to the `set` lines to make the result print e.g. `set(sample_file, j = "new_ra", value = ra)[],`